### PR TITLE
Don't cast an int * to a size_t *

### PR DESCRIPTION
**Pull request by mattcaswell**
If `sizeof(int) != sizeof(size_t)` this may not work correctly.

Fixes a Coverity issue.

### CODE REVIEW

File: apps/speed.c [hunk 1]
```diff
@@ -600,6 +600,7 @@ typedef struct loopargs_st {
     unsigned char *buf2_malloc;
     unsigned char *key;
     unsigned int siglen;
+    size_t sigsize;
```
Comment (levitte) [inline apps/speed.c]: Why do you keep siglen?  Should we be worried that we now have *two* fields expressing the same thing?

Comment (mattcaswell) [reply apps/speed.c]: Because its used lots of other places and needs be an int in those locations.

Comment (levitte) [reply apps/speed.c]: Ok...  as long as they aren't used interchangeably, it's fine with me

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**